evaluator/package_cache.go

```diff
@@ -0,0 +1,139 @@
+package evaluator
+
+import (
+	"app/base/database"
+	"app/base/utils"
+	"sync"
+)
+
+var memoryPackageCache *PackageCache
+
+type PackageCacheMetadata struct {
+	ID              int
+	NameID          int
+	Name            string
+	Evra            string
+	SummaryHash     []byte
+	DescriptionHash []byte
+}
+
+type PackageCache struct {
+	mtx          sync.RWMutex
+	data         []PackageCacheMetadata
```

Comment:
Do we really need to have the items in an array ? It seems it's never used

Reply:
I think it's for data itself to be stored somewhere. Other variables use pointers only.

Reply:
Exactly as Josca pointed out. I don't know golang data allocation strategy but I guess it's similar to C. You have to allocate/store data somewhere an then you can reference them via pointers.

Reply:
Well, and what if the storage array is resized ? That would mean dangling pointer. 

Go uses escape analysis. If a pointer to a struct is taken, it is allocated on the heap. If the struct list local temporary it uses the stack. 

Here you're just copying the struct from the heap, and storing the copy in an array:

https://github.com/RedHatInsights/patchman-engine/pull/661/files#diff-26fcc3d5a4f917e2d17fc4f92d45d4bc97904fcd69a9730369ca1f06f5adc126R128

Reply:
ok, removed redundant data array.